extension/libs/turndown.js

```diff
@@ -75,7 +75,47 @@ var TurndownService = (function () {
       tagNames.some(function (tagName) {
         return node.getElementsByTagName(tagName).length
       })
-    )
+    );
+  }
+
+  function inferAlt(img) {
+    // 1) Keep existing alt
+    const existing = img.getAttribute("alt");
+    if (existing && existing.trim()) {
+      return existing.trim();
+    }
+
+    // 2) Pull from a <figcaption> if inside a <figure>
+    const figcap = img.closest("figure")?.querySelector("figcaption");
+    if (figcap && figcap.textContent.trim()) {
+      return figcap.textContent.trim();
+    }
+
+    // 3) filename fallback & special “avatar” case
+    try {
+      const url = new URL(img.src, location.href);
```

Comment:
nit: I see we are using the global location object here. This may cause the code to fail in non-browser environments (like Node.js testing or server-side rendering). 

Should we consider adding a null check or providing a fallback? I am not sure this will be super relevant for extension though. I will let you take the call.

```suggestion
      const baseUrl = typeof location !== 'undefined' ? location.href : '';
      const url = new URL(img.src, baseUrl);
```

Reply:
@jatinkrmalik as you pointed out this is a browser extension so it’s highly unlikely that it’ll run on non-browser environments. If any use case pops up in future then we can modify the code.